### PR TITLE
Document dev cycle and feature freeze

### PR DESCRIPTION
**Pull request by joestringer**
See individual commits for more details:
- **docs: Fix organization docs flow**
- **docs: Remove unrelated paragraph for releases**
- **docs: Document feature release process**
- **docs: Document stable release timing**

### CODE REVIEW

File: Documentation/contributing/release/organization.rst [hunk 1]
```diff
@@ -31,5 +23,69 @@ for the next stable release. New patch releases for an existing stable version
...
+  ship functionality, and allows community members to get early access to
+  upcoming features to test and provide feedback. Prereleases may not be
+  published when a release candidate or final stable release is being
+  published.
+
+* Feature freeze: Around six weeks prior to a target feature release, the
+  ``main`` branch is frozen for new feature contributions. The goal of the
+  freeze is to focus community attention on stabilizing and hardening the
+  upcoming release by prioritizing bugfixes, documentation improvements, and
+  tests. In general, all new functionality that the community intends to
```
Comment (michi-covalent) [inline Documentation/contributing/release/organization.rst:51]:
```suggestion
  tests. In general, all new functionalities that the community intends to
```

**Review by michi-covalent — APPROVED** [review]
makes sense

**Review by aanm — APPROVED** [review]

File: Documentation/contributing/release/organization.rst [hunk 2]
```diff
@@ -7,20 +7,12 @@
...
-* :github-project:`GitHub release projects<>`
-
 Release Cadence
 ---------------
 
-New versions of Cilium are released on a cadence of around six months. Minor
-releases are typically designated by by incrementing the ``Y`` in the version
-format ``X.Y.Z``.
+New feature releases of Cilium are released on a cadence of around six months.
+Minor releases are typically designated by by incrementing the ``Y`` in the
```
Comment (ghost) [inline Documentation/contributing/release/organization.rst]:
```suggestion
Minor releases are typically designated by incrementing the ``Y`` in the
```

**Review by ghost — APPROVED** [review]
one nit, but looks good

File: Documentation/contributing/release/organization.rst [hunk 3]
```diff
@@ -31,5 +23,69 @@ for the next stable release. New patch releases for an existing stable version
...
+  ``main`` branch is frozen for new feature contributions. The goal of the
+  freeze is to focus community attention on stabilizing and hardening the
+  upcoming release by prioritizing bugfixes, documentation improvements, and
+  tests. In general, all new functionality that the community intends to
+  distribute as part of the upcoming release must land into the ``main`` branch
+  prior to this date. Any bugfixes, docs changes, or testing improvements can
+  continue to be merged as usual following this date.
+
+* Release candidates: Immediately following the feature freeze, the release
+  management team aims to publish a release candidate. These candidates should
```
Comment (aditighag) [inline Documentation/contributing/release/organization.rst]:
```suggestion
  management team aims to publish release candidates. These candidates should
```

**Review by aditighag — APPROVED** [review]

File: Documentation/contributing/release/organization.rst (hunk 1 above)
Comment (joestringer) [reply Documentation/contributing/release/organization.rst:51]: Interesting, `functionalities` sounds awkward to me. [Stackexchange discussion notes on the topic](https://english.stackexchange.com/questions/196590/functionality-vs-functionalities-are-both-correct-and-idiomatic) seem a bit mixed but generally come to the conclusion that either approach may be idiomatic in common English. :thinking:

File: Documentation/contributing/release/organization.rst (hunk 3 above)
Comment (joestringer) [reply Documentation/contributing/release/organization.rst]: :thinking: I guess the part about "Immediately" would contradict this suggestion, but given the title is `Release candidates`, it makes sense to adjust this wording a bit. I think I'll take your suggestion here and also drop the `Immediately` word.

**Pull request merged**